src/pywifes/splice.py

```diff
@@ -0,0 +1,330 @@
+        hdr_fluxvar["CRVAL1"] = blueSpec.header["CRVAL1"]
+        hdr_fluxvar["CDELT1"] = blueSpec.header["CDELT1"]
+        hdr_fluxvar["CTYPE1"] = "WAVE"
+        hdr_fluxvar["CUNIT1"] = "Angstrom"
+        hdr_fluxvar["BUNIT"] = "(count / Angstrom)^2"
+
+        hdu_fluxvar = fits.ImageHDU(data=fluxVar, header=hdr_fluxvar)
+        hdulist.append(hdu_fluxvar)
+
+        print("Saving spliced spectra")
+        hdulist.writeto(output, overwrite=True)
+        hdulist.close()
+
+
+
+
+def join_cubes(blue_path, red_path):
+
+    # Read red data and metadata
+    red_flux_cube = fits.getdata(red_path, 0, header=False)
+    red_fluxvar_cube = fits.getdata(red_path, 1, header=False)
+
+    redCRVAL1 = fits.getheader(red_path)["CRVAL3"]
+    redCDELT1 = fits.getheader(red_path)["CDELT3"]
+    redCRPIX1 = fits.getheader(red_path)["CRPIX3"]
+    redNAXIS1 = fits.getheader(red_path)["NAXIS3"]
+    red_wl = (np.arange(redNAXIS1) - redCRPIX1 + 1) * redCDELT1 + redCRVAL1
+    red_minWL = min(red_wl)
+    red_maxWL = max(red_wl)
+
+    # Read blue data and metadata
+    blue_flux_cube = fits.getdata(blue_path, 0, header=False)
+    blue_fluxvar_cube = fits.getdata(blue_path, 1, header=False)
+    blueCRVAL1 = fits.getheader(blue_path)["CRVAL3"]
+    blueCDELT1 = fits.getheader(blue_path)["CDELT3"]
+    blueCRPIX1 = fits.getheader(blue_path)["CRPIX3"]
+    blueNAXIS1 = fits.getheader(blue_path)["NAXIS3"]
+    blue_wl = (np.arange(blueNAXIS1) - blueCRPIX1 + 1) * blueCDELT1 + blueCRVAL1
+    blue_minWL = min(blue_wl)
+    blue_maxWL = max(blue_wl)
+
+
+
+    nwl = int((redCRVAL1 + redCDELT1 * redNAXIS1 - blueCRVAL1) / blueCDELT1)
+    wl = (np.arange(nwl) - blueCRPIX1 + 1) * blueCDELT1 + blueCRVAL1
+
+    # One does not need to interplolate the blue spectra if the waveelength is
+    # set to the blue spectrum. However the code is robust to this.
+
+    AB = _A_lanczos(blue_wl, wl, 3).tocsr()
+    AR = _A_lanczos(red_wl, wl, 3).tocsr()
+    
+
+    wave_dim, y_dim, x_dim  = np.shape(red_flux_cube)
+    wave_dim = len(wl)
+
+    flux_cube = np.zeros((wave_dim, y_dim, x_dim))
+    fluxvar_cube = np.zeros((wave_dim, y_dim, x_dim))
+
+
+
+    # Run over every point (i,j) in the spatial plane
+    for i in range(y_dim):
+        for j in range(x_dim):
+
+            red_flux =  red_flux_cube[:, i, j]
+            red_fluxvar =  red_fluxvar_cube[:, i, j]
+
+            blue_flux =  blue_flux_cube[:, i, j]
+            blue_fluxvar =  blue_fluxvar_cube[:, i, j]
+
+            flux_R = np.array(AR * red_flux).ravel()
+            flux_B = np.array(AB * blue_flux).ravel()
+
+            # Two sparse diagonal matrices
+            diag_R = sp.dia_matrix(([red_fluxvar], [0]), shape=[redNAXIS1, redNAXIS1])
+            diag_B = sp.dia_matrix(([blue_fluxvar], [0]), shape=[blueNAXIS1, blueNAXIS1])
+
+            fluxvar_B = np.array((AB * diag_B * AB.T).sum(axis=1)).ravel()
+            fluxvar_R = np.array((AR * diag_R * AR.T).sum(axis=1)).ravel()
```

Comment:
You could group these a little differently I think.

I'd go for 
```py
            red_flux =  red_flux_cube[:, i, j]
            flux_R = np.array(AR * red_flux).ravel()

            red_fluxvar =  red_fluxvar_cube[:, i, j]
            diag_R = sp.dia_matrix(([red_fluxvar], [0]), shape=[redNAXIS1, redNAXIS1])
            fluxvar_R = np.array((AR * diag_R * AR.T).sum(axis=1)).ravel()

            blue_flux =  blue_flux_cube[:, i, j]
            flux_B = np.array(AB * blue_flux).ravel()

            # Two sparse diagonal matrices
            blue_fluxvar =  blue_fluxvar_cube[:, i, j]
            diag_B = sp.dia_matrix(([blue_fluxvar], [0]), shape=[blueNAXIS1, blueNAXIS1])
            fluxvar_B = np.array((AB * diag_B * AB.T).sum(axis=1)).ravel()
```
because in each of those, only the final line is used further down the function. 

You *could* then go and make each of those groups its own function call, but it's not really necessary.